slice/internal/webhooks/jobset_webhook.go

```diff
@@ -55,22 +61,55 @@ func (r *JobSetWebhook) Default(ctx context.Context, obj runtime.Object) error {
 	log := ctrl.LoggerFrom(ctx).WithName("jobset-accelerator-gke-webhook")
 	log.V(5).Info("Applying defaults")
 
-	// handle webhooks for Kueue related JobSets
-	cq, ok := jobSet.Labels[kueueconstants.QueueLabel]
-	if !ok || cq == "" {
+	if jobSet.Labels[kueueconstants.QueueLabel] == "" {
 		return nil
 	}
 
 	for i, rj := range jobSet.Spec.ReplicatedJobs {
+		tpuTopology := rj.Template.Spec.Template.Annotations[TPUTopologyAnnotation]
+		tpuAccelerator := rj.Template.Spec.Template.Spec.NodeSelector[TPUAcceleratorLabel]
+
+		if tpuTopology == "" || tpuAccelerator == "" {
+			continue
+		}
+
 		if rj.Template.Annotations == nil {
 			rj.Template.Annotations = make(map[string]string)
 		}
-		rj.Template.Annotations[PodSetRequiredTopologyAnnotation] = AnnotationValueTBD
-		rj.Template.Annotations[PodSetSliceRequiredTopologyAnnotation] = AnnotationValueTBD
-		rj.Template.Annotations[PodSetSliceSizeAnnotation] = AnnotationValueTBD
+
+		rj.Template.Annotations[kueuealpha.PodSetRequiredTopologyAnnotation] = rj.Template.Spec.Template.Annotations[TPUBlockAnnotation]
+		rj.Template.Annotations[kueuealpha.PodSetSliceRequiredTopologyAnnotation] = rj.Template.Spec.Template.Annotations[TPUSubBlockAnnotation]
+
+		podSetSliceSize, err := podSetSliceSize(tpuTopology, rj.Template.Spec.Parallelism)
+		if err != nil {
+			return err
+		}
+		rj.Template.Annotations[kueuealpha.PodSetSliceSizeAnnotation] = fmt.Sprint(podSetSliceSize)
 
 		jobSet.Spec.ReplicatedJobs[i] = rj
 	}
 
 	return nil
 }
+
+func podSetSliceSize(tpuTopology string, parallelism *int32) (int32, error) {
+	dimensions := strings.Split(tpuTopology, "x")
+	if len(dimensions) < 2 || len(dimensions) > 3 {
```

Comment:
len() != 3?

Reply:
I think it is possible to set 2D dimension (2x2), no?